exasol/toolbox/util/dependencies/poetry_dependencies.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import re
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+import tomlkit
+from pydantic import (
+    BaseModel,
+    model_validator,
+)
+from tomlkit import TOMLDocument
+
+from exasol.toolbox.util.dependencies.shared_models import Package
+
+
+class PoetryGroup(BaseModel):
+    name: str
+    toml_section: Optional[str]
+
+    class Config:
+        frozen = True
+
+
+TRANSITIVE_GROUP = PoetryGroup(name="transitive", toml_section=None)
+
+
+class PoetryDependency(Package):
+    name: str
+    version: str
```

Comment:
Would it be better to use https://packaging.pypa.io/en/latest/version.html#packaging.version.Version here?